include/mxnet/lib_api.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file lib_api.h
+ * \brief APIs to interact with libraries
+ */
+#ifndef MXNET_LIB_API_H_
+#define MXNET_LIB_API_H_
+
+#include <cstdlib>
+
+#define INITIALIZE_STR "initialize"
+typedef int (*initialize_t)(int);
+
+extern "C" {
+    /*
+    Function: initialize
+    Parameters:
+    - int : MXNet version passed to library
```

Comment:
Do we need to document the format of version number?

Reply:
Good idea @junrushao1994. Maybe something like this:

```
/*! \brief version number is defined as:
*  MXNET_VERSION = (MXNET_MAJOR*10000 + MXNET_MINOR*100 + MXNET_PATCH)
*/ 
```

Reply:
@zheng-da https://github.com/apache/incubator-mxnet/blob/master/include/mxnet/base.h#L73-L82

Reply:
Added.